planner/core/exhaust_physical_plans.go

```diff
@@ -2387,7 +2387,11 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
 		}
 	} else if p.JoinType.IsSemiJoin() {
 		preferredBuildIndex = 1
-		fixedBuildSide = true
+		// MPPOuterJoinFixedBuildSide default value is false
+		// use MPPOuterJoinFixedBuildSide here as a way to disable using left table as build side
+		if !p.ctx.GetSessionVars().MPPOuterJoinFixedBuildSide && p.children[1].statsInfo().Count() > p.children[0].statsInfo().Count() {
+			preferredBuildIndex = 0
+		}
```

Comment:
Still need to set `fixedBuildSide` to `true` if `p.ctx.GetSessionVars().MPPOuterJoinFixedBuildSide` is true?